File: examples/hello-claude-agent-sdk-js/src/agent.js
import { query } from "@anthropic-ai/claude-agent-sdk";

export async function agent(prompt = "Introduce yourself.") {
  const text = [];

  for await (const message of query({ prompt })) {
    if ("result" in message && typeof message.result === "string") {
      text.push(message.result);
    }
  }

  return text.join("\n").trim();
}

File: examples/hello-claude-agent-sdk-python/agent.py
from __future__ import annotations

from claude_agent_sdk import query


async def agent(prompt: str = "Introduce yourself.") -> str:
    text: list[str] = []

    async for message in query(prompt=prompt):
        result = getattr(message, "result", None)
        if isinstance(result, str):
            text.append(result)

    return "\n".join(text).strip()
